src/ucp/proto/proto_multi.inl

```diff
@@ -48,16 +48,23 @@ ucp_proto_multi_max_payload(ucp_request_t *req,
                             const ucp_proto_multi_lane_priv_t *lpriv,
                             size_t hdr_size)
 {
-    size_t scaled_length =
-            ucp_proto_multi_scaled_length(lpriv, req->send.state.dt_iter.length);
-    size_t max_payload   = ucs_min(scaled_length, lpriv->max_frag - hdr_size);
+    size_t length   = req->send.state.dt_iter.length;
+    size_t max_frag = lpriv->max_frag - hdr_size;
+    size_t max_payload;
+
+    /* Do not split very small sends to chunks, it's not worth it, and
+       generic datatype may not be able to pack to a smaller buffer */
+    if (length < UCP_MIN_BCOPY) {
+        return max_frag;
+    }
 
+    max_payload = ucs_min(ucp_proto_multi_scaled_length(lpriv, length),
+                          max_frag);
     ucs_assertv(max_payload > 0,
-                "length=%zu weight=%.2f scaled_length=%zu max_frag=%zu "
-                "hdr_size=%zu",
+                "length=%zu weight=%zu%% lpriv->max_frag=%zu hdr_size=%zu",
                 req->send.state.dt_iter.length,
-                lpriv->weight / (double)UCS_BIT(UCP_PROTO_MULTI_WEIGHT_SHIFT),
-                scaled_length, lpriv->max_frag, hdr_size);
+                ucp_proto_multi_scaled_length(lpriv, 100), lpriv->max_frag,
```

Comment:
why 100?

Reply:
this prints percentage of data on this lane so it's 100*ratio